### PR TITLE
Add mkrc support for paths with spaces

### PR DESCRIPTION
**Pull request by fgandellini**
Setting temporarily IFS to "\n\b" it is possible make that mkrc support paths with spaces.
Let me know what you think about this... :)

### CODE REVIEW

**Comment by mike-burns** [comment]
Oh this is wild and exciting.

For those wondering "why `\b`?": https://stackoverflow.com/questions/16831429/when-setting-ifs-to-split-on-newlines-why-is-it-necessary-to-include-a-backspac

Messing with IFS frightens me, but it might be the fix we need.

Once a file with spaces is linked, can rcup(1) handle it?

**Comment by fgandellini** [comment]
You're right, this is not a clean solution, surely there is a better way to implement it, but this could be a first implementation.

From my tests also rcup(1) seems to work properly. I would appreciate if you could take a test you too in order to check it.

Thanks!

File: bin/mkrc.in [hunk 1]
```diff
@@ -84,6 +84,9 @@ fi
 
 files=$@
 
+PREVIFS=$IFS
+IFS=$(echo -en "\n\b")
```
Comment (pbrisbin) [inline bin/mkrc.in]: `echo`, especially with options, especially with these options, is a "portability train wreck":

- http://wiki.bash-hackers.org/commands/builtin/echo
- http://cfajohnson.com/shell/cus-faq.html#Q0b
- http://www.in-ulm.de/~mascheck/various/echo+printf/

`printf "\n\b"` would probably work, but I wonder if the simpler

````sh
IFS="
"
```

(as mike linked to in his comment) would be best.

**Comment by pbrisbin** [comment]
Note that this only solves filenames with spaces, not any other unusual characters which are totally valid. That said, it's probably a worthwhile 90%-solution.

File: bin/mkrc.in [hunk 2]
```diff
@@ -122,3 +125,5 @@ for file in $files; do
   $VERBOSE "Linking..."
   $INSTALL -d "$DOTFILES_DIR" -t "${tag:--}" $install_args "$dotless"
```
Comment (mike-burns) [inline bin/mkrc.in]: I worry about what changing `IFS` does to `$INSTALL` aka `rcup`. It iterates over space-separated things, such as `DOTFILES_DIRS`.

Would you mind adding a test for mkrc with multiple `DOTFILES_DIRS`?

**Comment by gfontenot** [comment]
Is there anything I can do to move this forward? I'd really like to have this fix available.

**Comment by mike-burns** [comment]
@gfontenot can you try this patch and make sure it works, including with multiple `DOTFILES_DIRS`?

**Comment by fgandellini** [comment]
Hi all! Sorry for the delay :(
I hope to write a test as soon as possible, but I doubt that it works with the current implementation, I'll keep you updated. 
Thanks!

**Comment by mike-burns** [comment]
I learned something about the Debian shell today:

```
~% dash
$ cd /dev
$ L='ls w*'
$ $L
watchdog  watchdog0  watchdog1
$ IFS="
> "
$ $L
dash: 6: ls w*: not found
$ 
```

I am seeing this in the tests too (from `test/mkrc-handle-spaces.t.log`):

```
-  '*/.dotfiles/file\ with\ spaces' -> '*/.file\ with\ spaces' (glob)
+  /home/mike/rcm/test/../bin/mkrc: line 127: rcup -v: command not found
```

I can also reproduce this on OpenBSD's sh.

**Comment by tofagerl** [comment]
Any updates on this? Even if it doesn't work on dash or sh, if it works on bash and zsh safely, I would consider using this branch since I keep having to manually work around this when trying to put anything in macOS' `~/Library/Application Support/`in my dotfiles...

**Comment by mike-burns** [comment]
Closing this PR: it had potential, but ultimately was unportable. I'd love to find a portable solution to this; please submit a PR if you find one.

**Comment by fgandellini** [comment]
Hi all, thanks for closing Mike!
I confirm that this fix doesn't work.
I was trying to implement the spaces management in the various scripts (in my opinion the management should be done in `lsrc`, `mkrc` and `rcup`) but it seems not so easy to do with bash scripting. I had been able to make `mkrc` work but not the other scripts...
Surprisingly for me the space management is very hard in bash! :(
If I'll find a way to manage it I'll open you a PR. Thanks again!

**Pull request closed without merging**